src/zenml/zen_stores/schemas/model_schemas.py

```diff
@@ -151,9 +155,13 @@ def update(
         Returns:
             The updated `ModelSchema`.
         """
+        from zenml.utils.tag_utils import create_links, remove_links
+
         for field, value in model_update.dict(exclude_unset=True).items():
-            if field == "tags":
-                setattr(self, field, json.dumps(value))
+            if field == "add_tags":
+                create_links(value, self.id, TaggableResourceTypes.MODEL)
+            elif field == "remove_tags":
+                remove_links(value, self.id)
```

Comment:
This is problematic since `create_links` and `remove_links` use the `Client()` under the hood, but the client should never be initialized like this inside the server since that leads to all sorts of unexpected behavior.

Instead, simply write those links to the DB directly here (or in the corresponding ZenStore method if it's not possible to do it inside the schema's `update()` function)

Reply:
Reworked to use SQL Store instance instead